pkg/sentry/platform/kvm/machine_arm64_unsafe.go

```diff
@@ -73,40 +77,44 @@ func updateVectorTable() {
 	}
 
 	toLocation := fromLocation + offset
-	page := getPageWithReflect(toLocation)
-	if err := syscall.Mprotect(page, syscall.PROT_READ|syscall.PROT_WRITE|syscall.PROT_EXEC); err != nil {
+	pageLen := int(syscall.Getpagesize())
+
+	// 1st page.
+	page1 := getPageWithReflect(toLocation, pageLen)
+	if err := syscall.Mprotect(page1, syscall.PROT_READ|syscall.PROT_WRITE|syscall.PROT_EXEC); err != nil {
 		panic(err)
 	}
-
-	page = getPageWithReflect(toLocation + 4096)
-	if err := syscall.Mprotect(page, syscall.PROT_READ|syscall.PROT_WRITE|syscall.PROT_EXEC); err != nil {
+	// 2nd page.
+	page2 := getPageWithReflect(toLocation+uintptr(pageLen), pageLen)
+	if err := syscall.Mprotect(page2, syscall.PROT_READ|syscall.PROT_WRITE|syscall.PROT_EXEC); err != nil {
 		panic(err)
 	}
 
 	// Move exception-vector-table into the specific address.
```

Comment:
Thanks for doing this. I really appreciate the effort to clean up the exemptions! I'll merge this if you like, but I do think this code could actually be improved in two ways (to align with the purpose of the nogo check).

1) Consider doing this from the ring0 package in an init() function. (Since this seems to be the wrong package, since it leaks tons of detail and makes a lot of assumptions about ring0.Vectors.)

2) Make it easy to audit and validate the actual unsafe code by just avoiding unsafe arithmetic directly, and instead relying on creating easily checkable slices, and then relying on bounds checking. I would argue that the current change is just tweaking the unsafe math a bit in order to satisfy the analyzer :)

For example, you could do something like this:

```
const (
   instSize = unsafe.Sizeof(uint32(0))
   vectors = 0x200
)

func unsafeInstructions(addr, length uintptr) (slice []uint32) {
        hdr := (*reflect.SliceHeader)(unsafe.Pointer(&slice))
        hdr.Data = addr
        hdr.Len = length/instSize
        hdr.Cap = length/instSize
        return slice
}

func init() {
    rewriteVectors()
}

func rewriteVectors() {
   ... mprotect, etc. ...

   // Map the vector code.
   vectorsBegin = reflect.ValueOf(Vectors).Pointer()
   vectorsEnd = FindEndAddress(memcpyBegin)
   vectors := unsafeSlice(vectorsBegin, vectorsEnd-vectorsBegin) // Now a []uint32.

   // From here on... all strictly unsafe code is done! Need only audit unsafeSlice and the above two lines.

  align := (1<<11)/instSize                 // By index, not bytes.
  offset := (vectorsBegin & (align-1)) // Ditto.
  if offset != 0 {
      offset = align - offset
  }
  for i := 1; i <= vectors; i++ {
      vectors[offset + vectors - i] = vectors[vectors-i]
  }
  for i := 0; i < vectors; i += 0x20 {
      vectors[offset + i] -= offset/4 // Adjust branch since instruction was moved forward.
  }

  // ... mprotect, etc. ..
}
```

Reply:
Done.
Please review it.